Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -225,6 +238,18 @@ PaintedContentsInfo::ContentsTypeDetermination PaintedContentsInfo::contentsType
     return m_contentsType;
 }
 
+#if HAVE(HDR_SUPPORT)
+RequestState PaintedContentsInfo::paintsHDRContentDetermination()
+{
+    if (m_hdrContent != RequestState::Unknown)
+        return m_hdrContent;
+
+    RenderLayer::PaintedContentRequest contentRequest;
```

Comment:
By making a new `PaintedContentRequest` here, you're missing the opportunity to share the same tree traversal that `PaintedContentsInfo::paintsContentDetermination() does. Use the existing `PaintedContentRequest` if you can.

Reply:
I created https://github.com/WebKit/WebKit/pull/40360 to address this issue.